models/sharky/solana/sharky_solana_events.sql

```diff
@@ -35,107 +30,333 @@ WITH sharky_txs AS (
+            {% else %}
+            True
+            {% endif %}
+    ), extend AS (
+        SELECT
+            call_block_date,
+            call_block_slot,
+            call_block_time,
+            call_tx_signer,
+            call_tx_id,
+            account_loan,
+            'Extend' AS evt_type,
+            account_escrow,
+            account_orderBook,
+            account_collateralMint,
+            ARRAY[CAST(ROW(call_data, call_account_arguments) AS ROW(data VARBINARY, account_arguments ARRAY<VARCHAR>))] AS sharky_instructions
+        FROM {{ source('sharky_solana', 'sharky_call_extendLoan') }}
+        WHERE
+            {% if is_incremental() %}
+            call_block_time >= DATE_TRUNC('day', NOW() - INTERVAL '7' DAY)
+            {% else %}
+            True
+            {% endif %}
+
+        UNION ALL
+        SELECT
+            call_block_date,
+            call_block_slot,
+            call_block_time,
+            call_tx_signer,
+            call_tx_id,
+            account_loan,
+            'Extend' AS evt_type,
+            account_escrow,
+            account_orderBook,
+            account_collateralMint,
+            ARRAY[CAST(ROW(call_data, call_account_arguments) AS ROW(data VARBINARY, account_arguments ARRAY<VARCHAR>))] AS sharky_instructions
+        FROM {{ source('sharky_solana', 'sharky_call_extendLoanEscrow') }}
+        WHERE
+            {% if is_incremental() %}
+            call_block_time >= DATE_TRUNC('day', NOW() - INTERVAL '7' DAY)
+            {% else %}
+            True
+            {% endif %}
+
+        UNION ALL
+        SELECT
+            call_block_date,
+            call_block_slot,
+            call_block_time,
+            call_tx_signer,
+            call_tx_id,
+            account_loan,
+            'Extend' AS evt_type,
+            account_escrow,
+            account_orderBook,
+            account_collateralMint,
+            ARRAY[CAST(ROW(call_data, call_account_arguments) AS ROW(data VARBINARY, account_arguments ARRAY<VARCHAR>))] AS sharky_instructions
+        FROM {{ source('sharky_solana', 'sharky_call_extendLoanV3') }}
+        WHERE
+            {% if is_incremental() %}
+            call_block_time >= DATE_TRUNC('day', NOW() - INTERVAL '7' DAY)
+            {% else %}
+            True
+            {% endif %}
+
+), with_amount AS (
+    SELECT
+       events.*,
+       CASE
+            -- For take and rescind instructions we can take the exact amount available in the escrow account
+            WHEN (
+                    events.evt_type = 'Take'
+                    OR events.evt_type = 'Rescind'
+                ) THEN abs(st.post_balances[array_position(st.account_keys, events.account_escrow)] - st.pre_balances[array_position(st.account_keys, events.account_escrow)] )
+            ELSE (abs(st.post_balances[1] - st.pre_balances[1]))
+       END AS amount_raw
+    FROM  {{ source('solana','transactions') }} st
```

Comment:
Is it possible to avoid the join with solana.transactions here? Why is it needed?

Reply:
We need to compute the balance change of either the tx signer or the escrow account